### PR TITLE
use mvnw to setup maven to use and upgrade setup-java action to use the cache

### PR DESCRIPTION
**Pull request by olamy**

### CODE REVIEW

**Comment by olamy** [comment]
@TimMoore FYI the issue you had with 4.0.0-alpha is due to this https://issues.apache.org/jira/browse/MINVOKER-319
which will be fixed with the coming version of the plugin (should be released by the end of the week)
so we can wait few days I guess and it will be fixed

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -14,29 +14,24 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        maven: [3.3.1, 3.5.4, 3.6.3, 3.8.7, 3.9.0]
+        maven: [3.3.1, 3.5.4, 3.6.3, 3.8.7, 3.9.0, 4.0.0-alpha-4]
+        jdk: ["11"]
```
Comment (TimMoore) [inline .github/workflows/maven.yml:18]: Great, I've been thinking that we should add Java versions to the matrix and start cross-testing. Thanks for setting this up. I can add additional versions in a separate change once this is merged.

**Comment by TimMoore** [comment]
> @TimMoore FYI the issue you had with 4.0.0-alpha is due to this https://issues.apache.org/jira/browse/MINVOKER-319
> which will be fixed with the coming version of the plugin (should be released by the end of the week)
> so we can wait few days I guess and it will be fixed

Great, thanks for investigating. Would you rather wait, or split that change back out again? If it's all the same to you, I'd prefer to merge the mvnw and setup-java changes separately, and keep the change to add 4.0.0-alpha-4 to the matrix on a separate branch until it's ready.

**Comment by olamy** [comment]
@TimMoore sure I just removed `4.0.0-alpha-4` from the list so the green should be green now.

**Comment by TimMoore** [comment]
Thanks, @olamy!

**Pull request merged**